spec/controllers/spree/api/line_items_controller_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'spec_helper'
+
+module Spree
+  describe Spree::Api::LineItemsController do
+    render_views
+
+    before do
+      stub_authentication!
+      Spree.user_class.stub :find_by_spree_api_key => current_api_user
+    end
+
+    def self.make_simple_data!
+      let!(:order) { FactoryGirl.create(:order, state: 'complete', completed_at: Time.now) }
+      let!(:line_item) { FactoryGirl.create(:line_item, order: order, unit_value: 500) }
+    end
+
+    #test that when a line item is updated, an order's fees are updated too
+    context "as an admin user" do
+      sign_in_as_admin!
+      make_simple_data!
+
+      context "as a line item is updated" do
+        it "apply enterprise fees on the item" do
+          line_item_params = { order_id: order.number, id: line_item.id, line_item: { id: line_item.id, unit_value: 520 }, format: :json}
+          controller.should_receive(:apply_enterprise_fees).and_return(true)
```

Comment:
HI @oeoeaio 
I'm a bit stuck on this one. I added .and_call_original to the stub. And I changed to order.should_receive(:update_distribution_charge!)

When I binding.pry spree_post :update, line_item_params the correct page is def returned but the test doesn't pass. order.update_distribution_charge! is never called. 

The tinterwebs seems to suggest that and_call_original is only available on partial doubles, suggesting to me that actually this isn't genuinely calling the original method and whole stack. 

So yeah, stuck. Do you have an example of using should_receive to test a method call within a stack? All the egs I've found in the code are same layer.

Hope you're enjoying your travels. Pity you didn't make it to Farm hack. Serverine from Greenhorns was there... and it was an wicked gathering :-) Any UK time planned?